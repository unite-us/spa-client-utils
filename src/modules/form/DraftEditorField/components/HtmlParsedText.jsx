import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import DOMPurify from 'dompurify';
import { get } from 'lodash';
import parse, { domToReact } from 'html-react-parser';

const options = {
  replace: (domNode) => {
    if (get(domNode, 'attribs.href')) {
      return (
        <a
          href={domNode.attribs.href}
          target="_blank"
          rel="noopener noreferrer"
        >
          {domToReact(domNode.children)}
        </a>
      );
    }
    return domNode;
  },
};

function HtmlParsedText(props) {
  const {
    className,
    value,
  } = props;

  return (
    <div className={classNames('html-parsed-text', className)}>
      { parse(DOMPurify.sanitize(value), options) }
    </div>
  );
}

HtmlParsedText.propTypes = {
  className: PropTypes.string,
  value: PropTypes.string.isRequired,
};

HtmlParsedText.defaultProps = {
  className: '',
};

export default HtmlParsedText;
