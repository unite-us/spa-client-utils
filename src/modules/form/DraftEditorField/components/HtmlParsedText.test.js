import React from 'react';
import renderer from 'react-test-renderer';
import HtmlParsedText from './HtmlParsedText';

describe('HtmlParsedText', () => {
  it('renders', () => {
    expect(renderer.create(
      <HtmlParsedText value="<div><p>Hello World</p> <a href='https://www.google.com'>yolo</a></div>" />,
    )).toMatchSnapshot();
  });
});
