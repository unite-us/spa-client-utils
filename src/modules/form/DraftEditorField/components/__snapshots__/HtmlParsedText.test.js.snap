// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HtmlParsedText renders 1`] = `
<div
  className="html-parsed-text"
>
  <div>
    <p>
      Hello World
    </p>
     
    <a
      href="https://www.google.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      yolo
    </a>
  </div>
</div>
`;
